apis/config/types.go

```diff
@@ -84,32 +84,36 @@ type MetricProviderSpec struct {
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// TrimaranArgs holds common arguments for trimaran plugins
+type TrimaranArgs struct {
+	metav1.TypeMeta
```

Comment:
I'd suggest embedding `MetricProvider` and `WatcherAddress` only.

Reply:
done